Part_1/Calculator/app/src/main/java/com/ensf409/calculator/Calculator.java

```diff
@@ -0,0 +1,223 @@
+	private char lastReceived;//keeps track of last received char on a ButtonPressed call
+	private boolean divZeroErr;//becomes true if a calculation attempts to divide by zero
+	
+	//initializes calculator
+	public Calculator(){
+		clear();
+		lastSum = 0;
+		lastReceived = 'C';
+		toDisplay = "";
+		divZeroErr = false;
+	}
+	//clears Lists of numbers & operations, sets afterDec and currentNumber to 0
+	//gets called by when 'C' or '=' is pressed and when initializing calculator
+	private void clear(){
+		numbers = new LinkedList<Double>();
+		operations = new LinkedList<Character>();
+		currentNumber = 0.0;
+		afterDec = 0;
+	}
+	//called from '=', calculates sum of stored numbers with respective operations and stores value in lastSum
+	//if a divide by zero is attempted, sets the divZeroErr to true 
+	private void calcSum(){
+		Double sum = numbers.pop();//set sum to the first number added, removed it from list
+		while(!numbers.isEmpty()){//while there are still numbers to pop, keep poppin'
+			char op = operations.pop();
+			Double num = numbers.pop();
+			if (op == '+')
+				sum += num;
+			else if (op == '-')
+				sum -= num;
+			else if (op == '*')
+				sum *= num;
+			else if (op == '/'){
+				//if a divide by zero is attempted, make note of it
+				if(num == 0)
+					divZeroErr = true;
+				sum /= num;
+			}
+			//should never reach this else statement if GUI is properly implemented
+			else
+				System.out.println("Operation not recognized...");
+		}
+		lastSum = sum;
+	}
+	//first version of overloaded ButtonPressed, accepts integer to build up currentNumber
+	public String ButtonPressed(int num){
+		//if divZeroErr has occurred, "lock out" method until 'C' has been pressed
+		if(divZeroErr)
+			return "Divide Zero Error";
+		double asDouble = (double)num;
+		//if decimal has been used on current number
+		if(afterDec!= 0){
+			//compute arithmetic for currentNumber
+			currentNumber += asDouble*Math.pow(10, (afterDec));
+			//this code is to produce a proper string toDisplay just in case 0's have been pressed after decimal place
+			int places = -1;
+			places *= (int)afterDec;
+			DecimalFormat df = new DecimalFormat("#.##");
+			df.setMaximumFractionDigits(places);
+			df.setMinimumFractionDigits(places);
+			toDisplay = df.format(currentNumber);
+			//decrement afterDec for proper arithmetic on next call of ButtonPressed
+			afterDec--;
+		}
+		//if decimal has not been used on currentNumber
+		else{
+			//compute arithmetic for currentNumber
+			currentNumber = (currentNumber*10) + asDouble;
+			toDisplay = Integer.toString((int)currentNumber.doubleValue());
+		}
+		//set lastReceived to 'N' notifying a number was entered last
+		lastReceived = 'N';
+		return toDisplay;
+	}
+	//first version of overloaded ButtonPressed, accepts a char, 4 possible char types 'O'(operation), 'C'(cancel), '=', or '.'
+	public String ButtonPressed(char ch){
+		//if '=' pressed
+		if(ch == '='){
```

Comment:
I think a `switch` statement might be better to use, since there are so many cases.